addons/web_editor/static/src/js/editor/snippets.options.js

```diff
@@ -5869,9 +5877,13 @@ const ImageHandlerOption = SnippetOptionWidget.extend({
     async _autoOptimizeImage() {
         await this._loadImageInfo();
         await this._rerenderXML();
-        this._getImg().dataset.resizeWidth = this.optimizedWidth;
-        await this._applyOptions();
-        await this.updateUI();
+        const img = this._getImg();
+        if (!['image/gif', 'image/svg+xml'].includes(img.dataset.mimetype)) {
+            img.dataset.mimetype = 'image/webp';
+            img.dataset.resizeWidth = this.optimizedWidth;
+            await this._applyOptions();
+            await this.updateUI();
+        }
```

Comment:
Can you explain this diff?
Why don't we want to call `_applyOptions` and `updateUI` anymore when we are not a `.GIF` or `.svg`?

Also, according to the forced `webp` mimetype, it looks like you want to only target this type of image, why not directly do that check in the condition?

Reply:
> Why don't we want to call _applyOptions and updateUI anymore when we are not a .GIF or .svg?

That was a mistake - there could be options to apply on those formats.

> Also, according to the forced webp mimetype, it looks like you want to only target this type of image, why not directly do that check in the condition?

It is to apply the recommended format and width change. Comment added. (The "Format" option does not exist for gif and svg)